### PR TITLE
Adding readout layers for use in grover

### PR DESCRIPTION
**Pull request by gusty1g**
# Pull Request Template

## Description

I have added a readout layer and corresponding tests for it. The readout layer for performing readout from a graph and used in models like Grover and other graph neural networks. Currently, it supports two ways for readout:
- readout via mean
- readout via a self-attention

In future, more readout operations can also be supported.

To perform readout on a batched graph, it accepts a list of embedding vectors. It also additionaly accepts a parameter `scope` which allows to group embedding of nodes/edges to a graph by mapping the start index and end index of the embedding list to a graph in the batch of graphs.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/readout.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+except ModuleNotFoundError:
+    raise ImportError('The module requires PyTorch to be installed')
+
+from deepchem.models.torch_models.attention import SelfAttention
+
+
+class GroverReadout(nn.Module):
+    """Performs readout on a batch of graph
+
+    The readout module is used for performing readouts on batched graphs to
```
Comment (rbharath) [inline deepchem/models/torch_models/readout.py:14]: Can you add a few sentences explaining how this is used in the Grover model?

**Review by rbharath — COMMENTED** [review]
Almost good to go, one minor comment

File: deepchem/models/torch_models/readout.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/models/torch_models/readout.py:14]: added docs in 521a676

**Review by rbharath — APPROVED** [review]
LGTM, please feel free to merge once CI looks good

**Pull request merged**